<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1166</width>
    <height>630</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">*{
	font: 12pt &quot;Segoe UI&quot;;
	color: rgb(240, 240, 240);
	background-color: rgb(90, 80, 80)
}

/*BUTTON*/
QPushButton{
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:1, stop:0 rgba(255, 166, 0, 255), stop:1 rgba(255, 85, 0, 255));
border-radius:15px;
font-whigth:bold;
}
QPushButton:hover{
	background-color:qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:1, stop:0 rgba(255, 200, 96, 255), stop:1 rgba(255, 128, 65, 255));
}
QPushButton:pressed{
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:1, stop:0 rgba(255, 166, 0, 240), stop:1 rgba(255, 85, 0, 240));
}

/*COMBOBOX*/
QComboBox {
	color: rgb(39, 39, 39);
    border-radius: 3px;
    padding: 1px 18px 1px 3px;
    min-width: 6em;
}
QComboBox:editable {
    background: white;
}
QComboBox:!editable, QComboBox::drop-down:editable {
     background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                 stop: 0 #E1E1E1, stop: 0.4 #DDDDDD,
                                 stop: 0.5 #D8D8D8, stop: 1.0 #D3D3D3);
}
QComboBox:!editable:on, QComboBox::drop-down:editable:on {
    background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                stop: 0 #D3D3D3, stop: 0.4 #D8D8D8,
                                stop: 0.5 #DDDDDD, stop: 1.0 #E1E1E1);
}
QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 25px;
border-top-right-radius:3px;
border-bottom-right-radius:3px;
	background-color:qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:1, stop:0 rgba(255, 166, 0, 255), stop:1 rgba(255, 85, 0, 255));
}

QComboBox::drop-down:hover {
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:1, stop:0 rgba(255, 200, 96, 255), stop:1 rgba(255, 128, 65, 255));
}


/*NUMBER PICKER*/
QSpinBox {
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:1, stop:0 rgba(255, 166, 0, 50), stop:1 rgba(255, 85, 0, 150));
border-radius:3px;
}
QSpinBox::up-button {
	background-color: qlineargradient(spread:pad, x1:0.507, y1:0, x2:0.532, y2:1, stop:0 rgba(255, 166, 0, 255), stop:1 rgba(255, 123, 0, 255));
}
QSpinBox::down-button {
	background-color: qlineargradient(spread:pad, x1:0.507, y1:0, x2:0.532, y2:1, stop:0 rgba(255, 123, 0, 255), stop:1 rgba(255, 85, 0, 255));
}
QSpinBox::up-button:hover, QSpinBox::down-button:hover {
	background-color: rgb(255, 170, 85);
}
QSpinBox::up-button:pressed,QSpinBox::down-button:pressed {
	background-color: rgb(200, 100, 0);
}

/*TABS*/
QTabWidget::pane { /* The tab widget frame */
    border-top: 2px solid rgb(255, 108, 0);
}
QTabWidget::tab-bar {
    left: 5px; /* move to the right by 5px */
}
QTabBar::tab {
	color: rgb(39, 39, 39);
    background: qlineargradient(spread:pad, x1:0.531925, y1:0.807, x2:0.523, y2:0, stop:0 rgba(187, 187, 187, 255), stop:1 rgba(232, 232, 232, 255)); 
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
    padding: 2px 15px;
margin-right:5px;
	border:1px solid rgb(90, 80, 80);
	border-bottom-color: rgb(255, 108, 0); /* same as the pane color */
}
QTabBar::tab:selected {
	color: rgb(240, 240, 240);
    background: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:1, stop:0 rgba(255, 166, 0, 255), stop:1 rgba(255, 85, 0, 255))
}
QTabBar::tab:hover {
	color: rgb(240, 240, 240);
    background: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:1, stop:0 rgba(255, 200, 96, 255), stop:1 rgba(255, 128, 65, 255));
}
/*SCROLL BAR*/
QScrollBar:vertical {
     border: none;
     background: rgb(70, 62, 62);
     width: 10px;
     margin: 20px 0;
 }
 QScrollBar::handle:vertical {
     background: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:1, stop:0 rgba(255, 170, 0, 255), stop:1 rgba(255, 85, 0, 255));
border-radius:3px;
     min-height: 20px;
 }
 QScrollBar::add-line:vertical {
     background: qlineargradient(spread:pad, x1:0.701781, y1:1, x2:0.298, y2:0.0113636, stop:0.00497512 rgba(213, 213, 213, 255), stop:1 rgba(224, 224, 224, 255));
	border-radius:3px;
     height: 20px;
     subcontrol-position: bottom;
     subcontrol-origin: margin;
 }

 QScrollBar::sub-line:vertical {
     background: qlineargradient(spread:pad, x1:0.701781, y1:1, x2:0.298, y2:0.0113636, stop:0.00497512 rgba(213, 213, 213, 255), stop:1 rgba(224, 224, 224, 255));
	border-radius:3px;
     height: 20px;
     subcontrol-position: top;
     subcontrol-origin: margin;
 }

 QScrollBar::add-page:vertical, QScrollBar::sub-page:vertical {
     background: none;
 }</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QTabWidget" name="tabWidget">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tab_3">
           <attribute name="title">
            <string>info</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <layout class="QHBoxLayout" name="infoLayout">
              <item>
               <layout class="QFormLayout" name="formLayout">
                <property name="labelAlignment">
                 <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
                </property>
                <property name="horizontalSpacing">
                 <number>15</number>
                </property>
                <item row="2" column="0">
                 <widget class="QLabel" name="label_6">
                  <property name="styleSheet">
                   <string notr="true">font-weight:500;</string>
                  </property>
                  <property name="text">
                   <string>Time function:</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QLabel" name="time_func">
                  <property name="text">
                   <string>-</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="0">
                 <widget class="QLabel" name="label_15">
                  <property name="styleSheet">
                   <string notr="true">font-weight:500;</string>
                  </property>
                  <property name="text">
                   <string>Memory function:</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="1">
                 <widget class="QLabel" name="memory_func">
                  <property name="text">
                   <string>-</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QLabel" name="label_13">
                  <property name="minimumSize">
                   <size>
                    <width>140</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>140</width>
                    <height>16777215</height>
                   </size>
                  </property>
                  <property name="styleSheet">
                   <string notr="true">font-size: 16pt ;
margin-bottom:10px;font-weight:400;</string>
                  </property>
                  <property name="text">
                   <string>Algorithm:</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QLabel" name="alg_type">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="minimumSize">
                   <size>
                    <width>0</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>1000</width>
                    <height>16777215</height>
                   </size>
                  </property>
                  <property name="styleSheet">
                   <string notr="true">font-size: 16pt ;
margin-bottom:10px;font-weight:500;</string>
                  </property>
                  <property name="text">
                   <string>Bubble Sort</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_8">
                  <property name="font">
                   <font>
                    <family>Segoe UI</family>
                    <pointsize>12</pointsize>
                    <weight>62</weight>
                    <italic>false</italic>
                    <bold>true</bold>
                   </font>
                  </property>
                  <property name="styleSheet">
                   <string notr="true">font-weight:500;</string>
                  </property>
                  <property name="text">
                   <string>Data generator order:</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QLabel" name="data_order_info">
                  <property name="text">
                   <string>Random order</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <widget class="Line" name="line_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>Additional info:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTextEdit" name="info">
              <property name="styleSheet">
               <string notr="true"/>
              </property>
              <property name="verticalScrollBarPolicy">
               <enum>Qt::ScrollBarAsNeeded</enum>
              </property>
              <property name="horizontalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOff</enum>
              </property>
              <property name="readOnly">
               <bool>true</bool>
              </property>
              <property name="html">
               <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:12pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="overwriteMode">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item alignment="Qt::AlignTop">
             <widget class="QPushButton" name="saveButton">
              <property name="minimumSize">
               <size>
                <width>180</width>
                <height>40</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Save to file</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab">
           <attribute name="title">
            <string>Plot (number/time)</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QCustomPlot" name="plot_time" native="true"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_2">
           <attribute name="title">
            <string>Plot (number/memory)</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QCustomPlot" name="plot_memo" native="true"/>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QLabel" name="label_5">
          <property name="styleSheet">
           <string notr="true">font: 16pt &quot;Segoe UI&quot;;
margin-bottom:10px;</string>
          </property>
          <property name="text">
           <string>Simulation</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Type of sorting:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="algorithm">
          <property name="minimumSize">
           <size>
            <width>189</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>360</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="maxVisibleItems">
           <number>11</number>
          </property>
          <item>
           <property name="text">
            <string>Bubble Sort</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Selection Sort</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Insertion Sort</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Quick Sort (Middle pivot)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Quick Sort (Random pivot)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Quick Sort (Median of Three pivot) </string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Merge Sort</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Shell Sort (Shell gap)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Shell Sort (Hibbard gap)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Shell Sort (Sedgewick gap)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Shell Sort (Ciura gap)</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Data generator type:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="data_order">
          <item>
           <property name="text">
            <string>Random order</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Ascending order</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Descending order</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Number of elements:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="num_elem">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>180</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="maximum">
           <number>100000000</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Step:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="step">
          <property name="maximumSize">
           <size>
            <width>180</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="maximum">
           <number>1000000</number>
          </property>
          <property name="value">
           <number>50</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Number of sorts:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="num_sorts">
          <property name="maximumSize">
           <size>
            <width>180</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="readOnly">
           <bool>false</bool>
          </property>
          <property name="showGroupSeparator" stdset="0">
           <bool>false</bool>
          </property>
          <property name="maximum">
           <number>100000</number>
          </property>
          <property name="stepType">
           <enum>QAbstractSpinBox::DefaultStepType</enum>
          </property>
          <property name="value">
           <number>30</number>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item alignment="Qt::AlignHCenter">
         <widget class="QPushButton" name="pushButton">
          <property name="minimumSize">
           <size>
            <width>180</width>
            <height>50</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>200</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">margin-bottom:10px;</string>
          </property>
          <property name="text">
           <string>START</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
